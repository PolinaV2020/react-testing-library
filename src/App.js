import React, {useState, useEffect} from "react";

const App = () => {
  const [data, setData] = useState(null);
  const [toggle, seToggle] = useState(false);
  const [inputValue, seInputValue] = useState(false);

  const onClick = () => seToggle(prev => !prev)

  useEffect(() => {
    setTimeout(() => setData({}), 100)
  }, [])

  return (
    <div>
      {toggle === true && <div data-testid="toggle-div">toggle</div>}
      <h1>hello world!</h1>
      <h2 data-testid="toggle-button">{inputValue}</h2>
      {data && <div style={{backgroundColor: "coral"}}>data</div>}
      <button data-testid="input-value" onClick={onClick}>Click me</button>
      <input onChange={(e) => seInputValue(e.target.value)} type="text" placeholder="Enter your text"></input>
    </div>
  );
}

export default App;
