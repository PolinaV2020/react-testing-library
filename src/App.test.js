import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe("App component", () => {

  test('Renders App elements', () => {
    render(<App />);
    const title = screen.getByText(/hello world/i);
    const button = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/Enter your text/i)
    expect(title).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(title).toMatchSnapshot();
  });

  test("Check that the element doesn't appear", () => {
    render(<App />);
    const text = screen.queryByText(/hidden text/i);
    expect(text).toBeNull();
  })

  test("Renders data text", async () => {
    render(<App />);
    const dataText = await screen.findByText(/data/i);
    expect(dataText).toBeInTheDocument();
    expect(dataText).toHaveStyle({backgroundColor: "coral"});
  })


  // test('Toggling', () => {
  //   render(<App />);
  //   const toggleButton = screen.getByTestId('toggle-button');
  //   expect(toggleButton).toBeInTheDocument();
  //   expect(screen.queryByTestId('toggle-div')).toBeNull();
  //   fireEvent.click(toggleButton);
  //   expect(screen.getByTestId('toggle-div')).toBeInTheDocument();
  // });
})





