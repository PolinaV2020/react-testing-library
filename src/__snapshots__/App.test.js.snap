// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders App elements 1`] = `
<h1>
  hello world!
</h1>
`;
